docs: Update cuda installation and support docs (#10909)
open-mpi/ompi: docs/networking/cuda.rst

      shell$ ./configure --prefix=/path/to/ucx-cuda-install --with-cuda=/usr/local/cuda --with-gdrcopy=/usr

      # Configure Open MPI this way
      shell$ ./configure --with-cuda=/usr/local/cuda --with-cuda-libdir=/usr/local/cuda/lib64/stubs/ --with-ucx=/path/to/ucx-cuda-install <other configure params>

#. Via internal Open MPI CUDA support

Regardless of which flavor of CUDA support (or both) you plan to use,
Open MPI should be configured using the ``--with-cuda=<path-to-cuda>``
and ``--with-cuda-libdir=<path-to-libcuda.so>`` configure options to
build CUDA support into Open MPI.

> Open MPI supports building with cuda libraries and running on systems
> without cuda libraries or hardware. In order to take advantage of
> this functionality, when compiling, you have to specify the cuda
> dependent components to be built as dso's using the

Reviewer: ```suggestion
Open MPI supports building with CUDA libraries and running on systems
without CUDA libraries or hardware. In order to take advantage of
this functionality, when compiling, you have to specify the CUDA
dependent components to be built as DOSs using the
```

Author: Done

``--enable-mca-dso=<comma-delimited-list-of-cuda-components.``
configure option.

> This affects the smcuda shared memory btl, rcache rgpusm, rcache
> gpusm, accelerator cuda, as well as the uct btl.

Reviewer: ```suggestion
This affects the ``smcuda`` shared memory and ``uct`` BTLs, as well as the ``rgpusm`` and ``gpusm`` rcache components.
```

Author: Done

An example configure command would look like the following:

   .. code-block:: sh

>       # Configure Open MPI this way

Author: Done

>       shell$ ./configure --with-cuda=/usr/local/cuda --with-cuda-libdir=/usr/local/cuda/lib64/stubs --enable-mca-dso=btl-smcuda,rcache-rgpusm,rcache-gpusm,accelerator-cuda <other configure params>

Reviewer: This is probably too wide.  Can you break it up into 2 lines, with a `\` continuation character?

Author: Done

/////////////////////////////////////////////////////////////////////////

OFI support for CUDA
---------------------

CUDA-aware support is present in OFI MTL.  When running CUDA-aware
Open MPI over Libfabric, the OFI MTL will check if there are any

> providers capable of handling gpu memory through the hmem related
> flags. If a cuda capable provider is available, the OFI MTL will
> directly send GPU buffers through libfabric's API after registering
> the memory. If there are no cuda capable providers available, the
> buffers will automatically be copied to host buffers before being
> transferred through libfabric's API.

Reviewer: ```suggestion
providers capable of handling GPU (or other accelerator) memory through the ``hmem``-related
flags. If a CUDA-capable provider is available, the OFI MTL will
directly send GPU buffers through Libfabric's API after registering
the memory. If there are no CUDA-capable providers available, the
buffers will automatically be copied to host buffers before being
transferred through Libfabric's API.
```

Author: Done
